### ISSUE TITLE
FNX-4710 ⁃ Restore content after crash

### ISSUE THREAD

**Issue by vesta0** [mozilla-mobile/fenix#1002]
As a user who has just experienced a browser crash, I want to see my pre-crash session and restore any crashed tabs so I can resume what I was doing with minimal friction

### Dependencies
see UX design here #1102 

### Acceptance Criteria (Added by PM)
I can restart Fenix and see the tabs from my pre-crash session
I can recover the tab or session
I will not enter a state where I cannot recover from the crash

**Comment by sblatz** [comment]
Please note that the current implementation leads to all tabs becoming "about:blank"

This seems to be [expected behavor](https://github.com/mozilla-mobile/android-components/issues/1333#issuecomment-440067815) and is exactly how it operates in Reference Browser currently. I will investigate to see if there is a GV fix in the works for this or if there's a way we can recover from this state.

**Comment by sblatz** [comment]
This is blocked on [this GV ticket](https://bugzilla.mozilla.org/show_bug.cgi?id=1530770)

**Comment by sblatz** [comment]
@vesta0, is this something we for sure want for MVP? The GV ticket has "not necessary for Fenix MVP" in its thread and just want to make sure it is correctly prioritized.

**Comment by sblatz** [comment]
Tagging @bbinto as Vesta is on PTO.

**Comment by vesta0** [comment]
@pocmo is this related to this [A-C issue](https://github.com/mozilla-mobile/android-components/issues/2571)?

**Comment by vesta0** [comment]
I believe we already have UX feedback here: #1102 @brampitoyo please confirm.

**Comment by brampitoyo** [comment]
@vesta0 Correct. I’ve given my feedback on #1102.

As @sblatz wrote:

> After some more work on this, it seems we actually don't have the ability to reload the tab just yet. I will just use a "close tab" button for now, and will add the "reload tab" when this functionality is added by the GV team.

Close tab is fine for now, but let’s make sure to ship reload as soon as it’s added on GV.

**Comment by vesta0** [comment]
@pocmo did the ability to restore tabs after crash just land in A-C? I saw a comment you made in another doc.

**Comment by sblatz** [comment]
@vesta0 I confirmed that the A-C issue you linked above did actually allow us to restore content. I believe this should be a priority, as we will **need to implement this anyway** in order to fix this [critical bug](https://github.com/mozilla-mobile/fenix/issues/1222)

**Comment by brampitoyo** [comment]
UX feedback is no longer necessary, because I’ve given mine 2 weeks ago. I’m going to unblock this issue now.

**Comment by sblatz** [comment]
@vesta0 if this is just referencing the ability to restore a single tab when it crashes, this work is complete.

**Comment by vesta0** [comment]
Thanks @sblatz ! I will add the QA needed label.

**Comment by sv-ohorvath** [comment]
> @vesta0 if this is just referencing the ability to restore a single tab when it crashes, this work is complete.

Verified that when a tab is crashed, only that tab is restored (the others are about:blank, I understand this is still expected).
After an app crash, al tabs are restored.

Would you like me to log a new issue for the about:blank tabs, for future work?

**Comment by sblatz** [comment]
Yes, @sv-ohorvath filling a new issue for the multiple tab case would be helpful :)
